packages/terra-avatar/src/common/AvatarUtils.jsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import classNames from 'classnames/bind';
+import TerraImage from 'terra-image';
+import styles from './Avatar.module.scss';
+
+const cx = classNames.bind(styles);
+
+const AVATAR_VARIANTS = {
+  USER: 'user',
+  FACILITY: 'facility',
+  SHARED_USER: 'shared-user',
+};
+
+const COLOR_VARIANTS = [
+  'one',
+  'two',
+  'three',
+  'four',
+  'five',
+  'six',
+  'seven',
+  'eight',
+  'nine',
+  'ten',
+];
+
+/**
+ * Generates a hash, based on a given string s.
+ * https://werxltd.com/wp/2010/05/13/javascript-implementation-of-javas-string-hashcode-method/
+ * @param {*} s
+ */
+const calculateHash = (s) => {
+  let hash = 0;
+  let char;
+
+  if (s.trim().length === 0) {
+    return hash;
+  }
+
+  /* eslint-disable no-bitwise */
+  for (let i = 0; i < s.length; i += 1) {
```

Comment:
I'm unfamiliar with calculating color hashes. Could you explain/comment a link of what this is doing? 🙃

Reply:
nooooo. I have a link above, but I can add some comments 👍

Reply:
I won't lie, I totally misses that looking at the no-bitwise loop 😅